lte/gateway/c/session_manager/LocalSessionManagerHandler.cpp

```diff
@@ -392,12 +334,112 @@ void LocalSessionManagerHandlerImpl::send_create_session(
           events_reporter_->session_create_failure(
               imsi, cfg.apn, cfg.mac_addr, failure_msg);
         }
-        LocalCreateSessionResponse resp;
-        resp.set_session_id(response.session_id());
-        response_callback(status, resp);
+        send_local_create_session_response(status, sid, cb);
       });
 }
 
+void LocalSessionManagerHandlerImpl::handle_create_session_cwf(
+    SessionMap& session_map, const LocalCreateSessionRequest& request,
+    const std::string& sid, SessionConfig cfg,
+    std::function<void(Status, LocalCreateSessionResponse)> cb) {
+  auto imsi       = request.sid().id();
+
+  // If there are no existing sessions for the IMSI, just create a new one
+  auto it = session_map.find(imsi);
+  if (it == session_map.end()) {
```

Comment:
why dont you just use the create session you have at line 384. The only difference is the log, isnt it?

Reply:
Yeah, I thought about that, but I think the intent is clearer if we just return early when the subscriber doesn't exist. I think I will bundle the recycling logic (line 357-379) into a helper function so that the function reads easier though.